### ISSUE TITLE
How to predict the softmax logits of the next word based on the previous word(or the id/softmax logits of the prev word) using your language model?

### ISSUE THREAD

**Issue by iamxiaoyubei** [tensorflow/lingvo#105]
Hi! Now I have the softmax logit of a word and I also can get its vocabulary id or word text. I want to  use this to predict the next word using your Language Model. 

But even though I looked at the code, I was still not sure what these [fetches](https://github.com/tensorflow/lingvo/blob/c8b6df48df825bbb51e8f59e0565228432e6fa82/lingvo/tasks/lm/model.py#L201) were and how to use them to get the softmax logit of the next word. 

So I hope you can give me some advice about how to predict the softmax logit of the next word based on the previous word(or the id/softmax logit of the prev word) using your language model?

Thank you!!

**Comment by jonathanasdf** [comment]
The xent_output (https://github.com/tensorflow/lingvo/blob/c8b6df48df825bbb51e8f59e0565228432e6fa82/lingvo/tasks/lm/model.py#L191) is this NestedMap (https://github.com/tensorflow/lingvo/blob/c8b6df48df825bbb51e8f59e0565228432e6fa82/lingvo/core/layers.py#L2483) so it seems xent_output.logits might be what you want. You need to add it yourself as a new fetch to the inference graph. 

The feeds that graph takes is just the text for now, but you can also add ids to the feeds to feed in ids instead.

**Comment by iamxiaoyubei** [comment]
Thank you! I have tried it, and it is indeed xent_output.logits: )

**Comment by zh794390558** [comment]
watch this.
